### ISSUE TITLE
Configuration of listeners and advertised-listeners

### ISSUE THREAD

**Issue by SteveWunderbar** [wurstmeister/kafka-docker#221]
In the current version the config of listeners and advertised-listeners cannot be set directly using environment variables since the settings are overwritten in `kafka-start.sh`. 
It would be much easier (and also more transparent) if one could configure the brokers with the values which are passed as env-variables. 

Effects: With the current version the brokers cannot connect to each other when using multiple ports (INSIDE, OUTSIDE, etc..) because of a broken config. Currently I run all the containers on the same host.

**What I configured with env-variables:**
```KAFKA_ADVERTISED_HOST_NAME: my.host
KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: INSIDE:PLAINTEXT,OUTSIDE:PLAINTEXT
KAFKA_ADVERTISED_PROTOCOL_NAME: OUTSIDE
KAFKA_PROTOCOL_NAME: INSIDE
KAFKA_ADVERTISED_PORT: my_outside_port
KAFKA_PORT: my_inside_port
KAFKA_BROKER_ID: -1
KAFKA_ZOOKEEPER_CONNECT: my.host:zookeeper_port
```

**What the resulting config should look like:**
```
advertised.host.name = null
advertised.listeners = OUTSIDE://my.host:xxxx,INSIDE://my.host:yyyy
advertised.port = null
```
...
```
host.name = 
inter.broker.listener.name = INSIDE
listener.security.protocol.map = INSIDE:PLAINTEXT,OUTSIDE:PLAINTEXT
listeners = OUTSIDE://:xxxx,INSIDE://:yyyy
```
...

**What it actually looks like:**
```
advertised.host.name = null
advertised.listeners = OUTSIDE://my.host:xxxx,INSIDE://my.host:yyyy
advertised.port = null
```
...
```
host.name = 
inter.broker.listener.name = INSIDE
listener.security.protocol.map = INSIDE:PLAINTEXT,OUTSIDE:PLAINTEXT
listeners =  #OUTSIDE://:9096,INSIDE://my.host:yyyy
```
...

I assume the problem with this is that Kafka tries to bind the port `yyyy` on `my.host` twice, which is not possible.

It would be nice to just use the env-variables like this (without any overwriting of `kafka-start.sh`):
```
KAFKA_ADVERTISED_LISTENERS: OUTSIDE://my.host:xxxx,INSIDE://my.host:yyyy
KAFKA_LISTENERS: OUTSIDE://:xxxx,INSIDE://:yyyy
KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: INSIDE:PLAINTEXT,OUTSIDE:PLAINTEXT
KAFKA_BROKER_ID: -1
KAFKA_ZOOKEEPER_CONNECT: my.host:zookeeper_port
```

Best regards,
SteveWunderbar

**Comment by tcjennings** [comment]
I agree in principal that you should be able to provide a complete listener config, and I have a PR ready for this use case, but I do not see a reproducible error condition here. What is it that goes wrong? I have not observed broker communication failure in the multi-broker docker swarms I operate.

**Comment by sscaling** [comment]
Fixed in #286
